packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -273,6 +274,11 @@ class TransferEngine<
     };
   }
 
+  async beforeTransfer(): Promise<void> {
+    await this.sourceProvider.beforeStreaming?.();
+    await this.destinationProvider.beforeStreaming?.();
```

Comment:
Should the names match to prevent confusion?

Reply:
you mean make the beforeTransfer ? beforeStreaming ?

Reply:
Yeah, I would make them both "beforeStreaming" I think (since "beforeTransfer" sounds like it might mean before we do any of the transfer process at all, even before bootstrap)

Reply:
To me it feels weird to have `transfer()` call several other things before it gets to calling `beforeTransfer()` but I can't think of a better alternative so I'm ok with it. In fact, I think beforeTransfer isn't the problem, but maybe renaming `transfer` itself could be renamed. But that's beyond the scope of this PR, so `beforeTransfer` here is good :)